community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnReadPageCursor.java

```diff
@@ -60,10 +62,24 @@ public boolean next() throws IOException
         }
         pin( nextPageId, false );
         currentPageId = nextPageId;
+        verifyContext();
         nextPageId++;
         return true;
     }
 
+    private void verifyContext()
+    {
+        if ( versionContext.lastClosedTransactionId() == Long.MAX_VALUE )
+        {
+            return;
+        }
+        if ( page.getLastModifiedTxId() > versionContext.lastClosedTransactionId() ||
```

Comment:
Would be very nice with human wording about what these conditions actually mean and how they play out.